components/merkledb/src/views/metadata.rs

```diff
@@ -12,34 +12,45 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use std::{borrow::Cow, cell::Cell, io::Error, mem};
+use std::{borrow::Cow, io::Error, mem};
 
 use byteorder::{LittleEndian, ReadBytesExt, WriteBytesExt};
 use enum_primitive_derive::Primitive;
 use failure::{self, ensure, format_err};
 use num_traits::FromPrimitive;
 use serde_derive::{Deserialize, Serialize};
 
-use crate::BinaryValue;
-
-use super::{IndexAccess, IndexAddress, View};
+use super::{IndexAddress, RawAccess, RawAccessMut, View};
+use crate::{validation::assert_valid_name, BinaryValue};
 
 /// Name of the column family used to store `IndexesPool`.
 const INDEXES_POOL_NAME: &str = "__INDEXES_POOL__";
 
-/// Type of the index stored in `IndexMetadata`.
+/// Type of an index supported by Exonum.
+///
 /// `IndexType` is used for type checking indexes when they are created/accessed.
 #[derive(Debug, Copy, Clone, PartialEq, Primitive, Serialize, Deserialize)]
 #[repr(u32)]
 pub enum IndexType {
+    /// Non-merkelized map index.
     Map = 1,
+    /// Non-merkelized list index.
     List = 2,
+    /// Singleton entry.
```

Comment:
I'm not sure if "signleton" is applicable here, since we can have multiple different entries.